### ISSUE TITLE
Memory not freed with MKL

### ISSUE THREAD

**Issue by andreasnoack** [JuliaLang/LinearAlgebra.jl#46]
I have some simulation code that accumulates memory allocation when Julia is build with MKL support but not when using OpenBLAS. It is a small simulation and still julia has allocated more than 19 gb (MKL) when done vs <0.5 gb (OpenBlas). Unfortunately I am unable to write a simple example that reproduces this. I wonder if it is related to

http://software.intel.com/en-us/articles/memory-leak-when-using-intel-mkl

My versioninfo is

```
Julia Version 0.3.0-prerelease+136
Commit 0d1beb8 (2013-11-24 09:35 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  WORD_SIZE: 64
  BLAS: libmkl_rt
  LAPACK: libmkl_rt
  LIBM: libopenlibm
```

**Comment by ViralBShah** [comment]
Cc: @ArchRobison - since you are the only guy around here we know from Intel.

**Comment by ArchRobison** [comment]
I'm not an MKL expert, but know where to find some :-)  What version of MKL are you using?  Is it possible to send the simulation code to me?  Does the problem go away if you set the environment variable MKL_DISABLE_FAST_MM=1 ?

**Comment by andreasnoack** [comment]
I am using MKL version 11.1.1. The simulation code is [here](https://gist.github.com/andreasnoackjensen/7655077) but you'll also need [this](https://github.com/andreasnoackjensen/Civecm.jl) package to run the code. I have tried with MKL_DISABLE_FAST_MM=1 but julia still allocates around 19 gb.

**Comment by ArchRobison** [comment]
I'm able to run the example now.  But it seems to be allocating only about 4.7 GB.  But I'm getting messages:

```
Warning: Possible conflict in library symbol dgetrs_
Warning: Possible conflict in library symbol dlaic1_
Warning: Possible conflict in library symbol dormqr_
Warning: Possible conflict in library symbol dscal_
Warning: Possible conflict in library symbol dpotrf_
Warning: Possible conflict in library symbol dpotri_
Warning: Possible conflict in library symbol dgemv_
```

so I'm not convinced yet that I'm really using MKL.  I've going to try comparing the behavior against a build of Julia that is definitely not using MKL.

**Comment by ViralBShah** [comment]
Are you on a mac?

**Comment by ArchRobison** [comment]
@andreasnoackjensen: how much real memory is on your system?  I'm on a 64-bit Linux box with 8 GB of real memory.

**Comment by andreasnoack** [comment]
My system is identical to yours. 64-bit Linux with 8 GB of real memory.

**Comment by ArchRobison** [comment]
Thanks for the info.  I have sufficient info now to study the problem.  So far I have two builds of Julia, one with MKL+ifort, and the other without.  The former uses 28x more memory than the latter. [Update: ratio seems closer to 20x now -- I was using "top" to informally measure memory consumption.] [Update: I was looking at the physical memory consumption.  The ratio of virtual memory consumption is about 50x.]

**Comment by ArchRobison** [comment]
I filed a bug report with the MKL team.  By the way, I noticed the MKL version was significantly faster than the default version in my latest runs.  Though the MKL version was using only 4.7 GB for me, much smaller than 19 GB, but still much larger than the 232 MByte in the default version.

**Comment by ViralBShah** [comment]
We should make sure that the same number of threads are being used in both cases. It is certainly possible that MKL can be faster - but just need to be sure. Thanks for filing the bug report.

**Comment by ArchRobison** [comment]
Good point.  I believe the MKL used by Julia is the parallel version -- I noticed it when trying Amplifier a few months ago.

**Comment by ArchRobison** [comment]
Update: Fix for memory leak is planned for MKL 11.1.2.  MKL team recommends rolling back to the MKL 11.0.5 release in the meantime.

**Comment by andreasnoack** [comment]
Great. Thank you for looking into this.

**Comment by andreasnoack** [comment]
I can confirm that the problem is solved in MKL 11.1.2. Thank you @ArchRobison.
